internal/security/fileprovider/provider_test.go

```diff
@@ -0,0 +1,420 @@
+	mockFileIoPerformer.On("OpenFileReader", configFile, os.O_RDONLY, os.FileMode(0400)).Return(strings.NewReader(`{"service1":{},"service2":{}}`), nil)
+	mockFileIoPerformer.On("OpenFileWriter", expectedService1File, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, os.FileMode(0600)).Return(&writeCloserBuffer{service1Buffer}, nil)
+	expectedService2Dir := filepath.Join(outputDir, "service2")
+	expectedService2File := filepath.Join(expectedService2Dir, outputFilename)
+	service2Buffer := new(bytes.Buffer)
+	mockFileIoPerformer.On("MkdirAll", expectedService2Dir, os.FileMode(0700)).Return(nil)
+	mockFileIoPerformer.On("OpenFileWriter", expectedService2File, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, os.FileMode(0600)).Return(&writeCloserBuffer{service2Buffer}, nil)
+
+	mockAuthTokenLoader := &MockAuthTokenLoader{}
+	mockAuthTokenLoader.On("Load", privilegedTokenPath).Return("fake-priv-token", nil)
+
+	expectedService1Policy := "{}"
+	expectedService2Policy := "{}"
+	expectedService1Parameters := makeMetaServiceName("service1")
+	expectedService2Parameters := makeMetaServiceName("service2")
+	mockSecretStoreClient := &MockSecretStoreClient{}
+	mockSecretStoreClient.On("InstallPolicy", "fake-priv-token", "edgex-service-service1", expectedService1Policy).Return(http.StatusNoContent, nil)
+	mockSecretStoreClient.On("InstallPolicy", "fake-priv-token", "edgex-service-service2", expectedService2Policy).Return(http.StatusNoContent, nil)
+	mockSecretStoreClient.On("CreateToken", "fake-priv-token", expectedService1Parameters, mock.Anything).
+		Run(func(args mock.Arguments) {
+			setCreateTokenResponse("service1", args.Get(2).(*interface{}))
+		}).
+		Return(http.StatusOK, nil)
+	mockSecretStoreClient.On("CreateToken", "fake-priv-token", expectedService2Parameters, mock.Anything).
+		Run(func(args mock.Arguments) {
+			setCreateTokenResponse("service2", args.Get(2).(*interface{}))
+		}).
+		Return(http.StatusOK, nil)
+
+	p := NewTokenProvider(mockLogger, mockFileIoPerformer, mockAuthTokenLoader, mockSecretStoreClient)
+	p.SetConfiguration(secretstoreclient.SecretServiceInfo{}, config.TokenFileProviderInfo{
+		PrivilegedTokenPath: privilegedTokenPath,
+		ConfigFile:          configFile,
+		OutputDir:           outputDir,
+		OutputFilename:      outputFilename,
+	})
+
+	// Act
+	err := p.Run()
+
+	// Assert
+	// - {OutputDir}/service1/{OutputFilename} w/proper contents
+	// - {OutputDir}/service2/{OutputFilename} w/proper contents
+	// - Correct policy for service1
+	// - Correct policy for service2
+	// - All other expectations met
+	assert.NoError(t, err)
+	mockFileIoPerformer.AssertExpectations(t)
+	mockAuthTokenLoader.AssertExpectations(t)
+	mockSecretStoreClient.AssertExpectations(t)
+	assert.Equal(t, expectedTokenFile("service1"), service1Buffer.Bytes())
+	assert.Equal(t, expectedTokenFile("service2"), service2Buffer.Bytes())
+}
+
+func setCreateTokenResponse(serviceName string, retval *interface{}) {
```

Comment:
Should `serviceName` be used? If not, remove `serviceName`.

Reply:
Removed.  Root cause is I though that I wanted to set the metadata property on the response.  In reality, I check that I set the metadata property on the call to the secretstore client and don't care about the actual response.